libs/execution_context/ebpf_core.c

```diff
@@ -399,6 +406,33 @@ _ebpf_core_protocol_map_update_element(_In_ const epf_operation_map_update_eleme
     return retval;
 }
 
+static ebpf_result_t
+_ebpf_core_protocol_map_update_element_with_handle(
+    _In_ const epf_operation_map_update_element_with_handle_request_t* request)
+{
+    ebpf_result_t retval;
+    ebpf_map_t* map = NULL;
+
+    retval = ebpf_reference_object_by_handle(request->map_handle, EBPF_OBJECT_MAP, (ebpf_object_t**)&map);
+    if (retval != EBPF_SUCCESS)
+        goto Done;
+
+    const ebpf_map_definition_t* map_definition = ebpf_map_get_definition(map);
+
+    if (request->header.length < (EBPF_OFFSET_OF(epf_operation_map_update_element_with_handle_request_t, data) +
+                                  map_definition->key_size + map_definition->value_size)) {
+        retval = EBPF_INVALID_ARGUMENT;
+        goto Done;
+    }
+
+    retval = ebpf_map_update_entry_with_handle(
+        map, request->data, request->data + map_definition->key_size, request->value_handle);
```

Comment:
If this function grabbed the struct ebpf_program* from the handle here and wrote the concatenated fd + pointer using ebpf_map_update_entry, then we could do away with ebpf_map_update_entry_with_handle function. program_array type maps will implement the same functions as other types. If you think this can simplify the code, please consider.

Reply:
for compatibility reasons, the map's value_size needs to not include the extra bytes, so we cannot pass extra bytes as part of the value itself